src/middleware/identity/smartManagement.js

```diff
@@ -1,12 +1,40 @@
 'use strict';
 
+const _ = require('lodash');
+
+const config = require('../../config');
 const errors = require('../../errors');
+const inventory = require('../../connectors/inventory');
+const queries = require('../../remediations/remediations.queries');
 
-module.exports = function (req, res, next) {
-    if (!req.entitlements.smart_management) {
-        return next(new errors.Forbidden());
-    }
+module.exports = async function (req, res, next) {
+    try {
+        let rhcConnectedSystems = false;
+        if (config.isMarketplace) {
+            const remediation = await queries.get(req.params.id, req.user.account_number, req.user.username);
```

Comment:
This is a costly check. There should be a fast lane for customers with `req.entitlements.smart_management`